blocks/algorithms/__init__.py

```diff
@@ -192,7 +202,9 @@ class GradientDescent(DifferentiableCostMinimizer):
 
     """
     def __init__(self, step_rule=None, gradients=None, known_grads=None,
-                 consider_constant=None, **kwargs):
+                 consider_constant=None,
+                 on_unused_sources='warn',
```

Comment:
You said it raises by default, but the code suggests that it just warns. I would raise by default, since this is the safest option and because this is the way it was done before. What do you think?